src/Castle.Core/DynamicProxy/ProxyGenerationOptions.cs

```diff
@@ -296,8 +293,61 @@ public override int GetHashCode()
 			result = 29*result + (Selector != null ? 1 : 0);
 			result = 29*result + MixinData.GetHashCode();
 			result = 29*result + (BaseTypeForInterfaceProxy != null ? BaseTypeForInterfaceProxy.GetHashCode() : 0);
-			result = 29*result + CollectionExtensions.GetContentsHashCode(AdditionalAttributes);
+			result = 29*result + GetAdditionalAttributesHashCode();
```

Comment:
`GetHashCode` does not have to be precise (only fast), so we could just use `AdditionalAttributes.Count` instead of summing the attributes' hash codes. I've not done so because there are some unit tests affected by it; someone in the past must have thought  it important that `options.GetHashCode` differ for different attributes.